xmtp_mls/src/client.rs

```diff
@@ -68,6 +81,99 @@ where
         }
     }
 
+    // TODO: Remove this and figure out the correct lifetimes to allow long lived provider
+    fn mls_provider(&self) -> XmtpOpenMlsProvider {
+        XmtpOpenMlsProvider::new(&self.store)
+    }
+
+    pub async fn register_identity(&self) -> Result<(), ClientError> {
+        // TODO: Mark key package as last_resort in creation
+        let last_resort_kp = self.identity.new_key_package(&self.mls_provider())?;
+        let last_resort_kp_bytes = last_resort_kp.tls_serialize_detached()?;
+
+        self.api_client
+            .register_installation(last_resort_kp_bytes)
+            .await?;
+
+        Ok(())
+    }
+
+    pub async fn top_up_key_packages(&self) -> Result<(), ClientError> {
```

Comment:
We have the concept of "one-time key packages" that can only be downloaded once, and then the server will delete them. They get used any time someone wants to invite you to a group. Because of their one-time nature, we need to periodically refill the supply so that new members can keep getting fresh key packages. This tops up the server's supply.

We were discussing today the possibility of removing the one-time key packages and going to a different model where key packages were re-used, but the client would replace the currently active key package every time it starts up. That may be easier to work with in our model.